docs/reference/networking.md

```diff
@@ -1,13 +1,89 @@
 # Virtlet networking principles
 
+The only supported configuration is the use of
+[CNI plugins](https://github.com/containernetworking/cni). Virtlet should work
+with any CNI plugin which is returning correct
+[Result](https://github.com/containernetworking/cni/blob/spec-v0.3.1/SPEC.md#result)
+in at least 0.3.0 format, according to CNI protocol specification.
+
+Virtlet assumes that selected CNI plugins will plug prepared for VM network
+namespace into own network using [veth](http://man7.org/linux/man-pages/man4/veth.4.html)
+or [SR-IOV](https://en.wikipedia.org/wiki/Single-root_input/output_virtualization)
+interfaces.
+
+For each configured veth interface Virtlet will prepare additional
+bridge (used later by internal DHCP server) used to connect this veth with
+[TAP](https://en.wikipedia.org/wiki/TUN/TAP) which will be then passed to
+hypervisor ([QEMU](https://www.qemu.org)) process.
+
+Each configured SR-IOV device will be removed from host kernel configuration
+and passed to hypervisor as host-passthrough PCI device preserving their
+hardware MAC addresses and optionanlly configured by plugin [VLAN](https://en.wikipedia.org/wiki/Virtual_LAN).
+
 # Supported CNI implementations
 
+Virtlet was tested and is working correctly with:
+
+* [Flannel](https://github.com/coreos/flannel)
+* [Calico](https://github.com/projectcalico/cni-plugin)
+* [Weave](https://github.com/weaveworks/weave)
+* [original](https://github.com/hustcat/sriov-cni) and [Intel fork](https://github.com/intel/sriov-cni) of SR-IOV
+* [combination](#multi-cni) of them behind [CNI-Genie](https://github.com/Huawei-PaaS/CNI-Genie)
+
 # Configuring using DHCP
 
+Information collected from CNI plugin is used by DHCP server accessible on each
+of configured for VM interfaces. That data is visible only for particular VM
+and is limited to serve configuration for particular interface, per interface.
+
+DHCP configuration is unavailable for [SR-IOV](#sriov) devices as they are
+passed directly to VM while fully excluded from host kernel configuration.
+
 # Configuring using Cloud-Init
 
-Not supported for persistent rootfs for now
+In parallel to passing network configuration through DHCP it is also passed to
+VM as a part of [Cloud-init](./cloud-init.md) data. That is the preffered
+source of configuration data for SR-IOV devices.
+Data configured by Virtlet is formed following
+[Network Config Version 1](https://cloudinit.readthedocs.io/en/latest/topics/network-config-format-v1.html)
+format.
+
+Note: Cloud-init network configuration is not supported for persistent rootfs
+for now.
+
+# <a name="multi-cni"></a> Setting up Multiple CNIs
```

Comment:
afair `<a name="multi-cni"></a>` is not valid markdown, instead there should be `[Setting up Multiple CNIs] (#multi-cni)` and then normally `# Setting up Multiple CNIs`

Reply:
We are using this syntax in several places already. You can take a look on preview of result file in https://github.com/ivan4th/virtlet-docs/blob/6f9e0940f88267e55feda9140f6f0debb6ebe202/docs/reference/networking.md - where this works correctly.

If you will insist - no problem, I can change that (and will do a follow up to change that in other places).

Reply:
I'd prefer that.

Reply:
Nevermind, let's keep it like that for now ;-)